invalid memory address at stmtsummary/v2.(*StmtSummary).Enabled

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x29326dd]


goroutine 88196 [running]:

github.com/pingcap/tidb/util/stmtsummary/v2.(*StmtSummary).Enabled(...)

	util/stmtsummary/v2/stmtsummary.go:156

github.com/pingcap/tidb/util/stmtsummary/v2.Enabled()

	util/stmtsummary/v2/stmtsummary.go:511 +0x9d

github.com/pingcap/tidb/executor.(*ExecStmt).SummaryStmt(0xc025bfb3b0, 0x1)

	executor/adapter.go:1854 +0xa5

github.com/pingcap/tidb/executor.(*ExecStmt).FinishExecuteStmt(0xc025bfb3b0, 0x5?, {0x0, 0x0}, 0x70?)

	executor/adapter.go:1469 +0x387

github.com/pingcap/tidb/executor.(*ExecStmt).CloseRecordSet(0xc025bfb3b0, 0x2f2c5254b?, {0x0?, 0x0?})

	executor/adapter.go:1542 +0x38

github.com/pingcap/tidb/executor.(*recordSet).Close(0xc0260f6dc0)

	executor/adapter.go:203 +0x67

github.com/pingcap/tidb/session.(*execStmtResult).Close(0xc0260df9b0)

	session/session.go:2416 +0x42

github.com/pingcap/tidb/parser/terror.Call(0xc02df2f820?)

	parser/terror/terror.go:306 +0x3b

github.com/pingcap/tidb/ddl.(*session).execute(0xc02c624bd0, {0x5421488, 0xc00022c000}, {0xc01705b930, 0xc8}, {0x4cae580, 0xd})

	ddl/ddl.go:1674 +0x44b

github.com/pingcap/tidb/ddl.(*ddl).getJob(0xc02cb4f480, 0x0?, 0x0?, 0xc02df2fb68)

	ddl/job_table.go:105 +0x175

github.com/pingcap/tidb/ddl.(*ddl).getReorgJob(0x53f94a0?, 0xc02df0c280?)

	ddl/job_table.go:155 +0x52

github.com/pingcap/tidb/ddl.(*ddl).loadDDLJobAndRun(0xc02cb4f480, 0xc02df2fdb8?, 0xc02d6d6cc0, 0xc02df2fe48)

	ddl/job_table.go:217 +0x139

github.com/pingcap/tidb/ddl.(*ddl).startDispatchLoop(0xc02cb4f480)

	ddl/job_table.go:202 +0x7e7

github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()

	util/wait_group_wrapper.go:154 +0x6b

created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run

	util/wait_group_wrapper.go:152 +0x8d

```

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_unit_test/detail/ghpr_unit_test/16678/pipeline
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/stmtsummary/v2/tests/table_test.go
+++ b/util/stmtsummary/v2/tests/table_test.go
@@ -515,9 +515,9 @@ func setupStmtSummary() {
 }
 
 func closeStmtSummary() {
-	stmtsummaryv2.GlobalStmtSummary.Close()
-	stmtsummaryv2.GlobalStmtSummary = nil
 	config.UpdateGlobal(func(conf *config.Config) {
 		conf.Instance.StmtSummaryEnablePersistent = false
 	})
+	stmtsummaryv2.GlobalStmtSummary.Close()
+	stmtsummaryv2.GlobalStmtSummary = nil
 }

